lambdas/data-migration1/src/providers.ts

```diff
@@ -59,14 +59,26 @@ export const migrateProviderRecord = async (
   providerKmsKeyId: string,
   knex: Knex
 ): Promise<void> => {
+  const providerPgModel = new ProviderPgModel();
+
   // Use API model schema to validate record before processing
   Manager.recordIsValid(dynamoRecord, schemas.provider);
 
-  const existingRecord = await knex<PostgresProviderRecord>('providers')
-    .where('name', dynamoRecord.id)
-    .first();
+  let existingRecord;
+
+  try {
```

Comment:
Nit: could this logic be DRY'd up?    Not needed for this PR but something to consider.

Reply:
Yeah, there's definitely a lot of duplication across the data-migration lambdas and tests generally. The `migrateXRecord` functions and `migrateXs` basically all do the same thing and could probably be DRY'd up quite a lot.

It would be nice to clean that up. Because we don't really have an obvious (please correct me) module that migration helpers could live in, I'm a little hesitant to do it here due to the size of the refactor (mostly tests) though.

Would it be worth doing separately? I get stuck thinking that these won't need to be extended further but I've never once been right about that assumption 😄

Reply:
Had a chat in Slack about this. There is definitely duplication here but _at this moment_ it seems like it would cost more time overall to address it as opposed to saving time by making these easier to extend/test. 

That doesn't mean we shouldn't refactor though. I'm just thinking we should address that if/when it comes up, mainly due to the sorta transient nature of these Lambdas.

Also, I get that you're referring to one block of code and I'm extrapolating to the whole Lambda 😄.  The catch with moving this one block to a shared module somewhere is that we don't really have a super obvious place for it so we'd likely need to create one. Totally doable but that pushes it out of the "just do it" mentality for me.